crates/network/src/lib.rs

```diff
@@ -50,8 +50,20 @@ pub struct BlockResponse<N: Network> {
     transactions: TransactionList<N::TransactionResponse>,
 }
 
+/// A receipt response.
+///
+/// This is distinct from [`TxReceipt`], since this is for JSON-RPC receipts.
+///
+/// [`TxReceipt`]: alloy_consensus::TxReceipt
+pub trait ReceiptResponse {
```

Comment:
I added this because consensus receipts do not have the contract address field, and they do not have the means to calculate it either. We cannot recalculate it ourselves where needed, since the receipt might be the result of a transaction where we do not know the nonce (e.g. a partially filled `send_transaction` signed by the node instead of us)

Reply:
Ah I see, ignore my previous comment. Let's add a note about this in the docs